data/src/test/java/com/linkedin/data/TestData.java

```diff
@@ -866,6 +866,23 @@ public void testCopy() throws CloneNotSupportedException
     }
   }
 
+  // Tests copy method in the presence of hash collisions in the data objects
+  @Test
+  public void testCopyHashCollisions() throws Exception
+  {
+    DataMap a = new DataMap();
+    a.put("b", new DataMap());
+    a.put("c", new DataMap());
+
+    // Use the next hashcode to cause collision.
+    a.getDataMap("c")._dataComplexHashCode = DataComplexHashCode.nextHashCode() + 1;
+    DataMap copy = a.copy();
+    assertNotNull(copy.get("b"));
+    assertNotNull(copy.get("c"));
+    assertNotSame(copy.get("b"), copy);
```

Comment:
what is the purpose of test the comparision of  "b"

Reply:
This is to check if the copy is not reusing same datamap object at different levels due to hash collision. (this was happening before the fix).

Reply:
But you didn't seem to create collision for “b"?

Reply:
yes, but still no harm in testing it? :)

Reply:
sure! Same thought. Thanks for confirmation